integrations/aws/main.py

```diff
@@ -0,0 +1,184 @@
+    sessions = _get_sessions()
+    async for session in sessions:
+        async for batch in batch_resources(kind, session, 'elbv2', 'describe_load_balancers', 'LoadBalancers', 'NextMarker'):
+            yield batch
+
+@ocean.on_resync(kind=ResourceKindsWithSpecialHandling.CLOUDFORMATION)
+async def resync_cloudformation(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    async for session in _get_sessions():
+        async for batch in batch_resources(kind, session, 'cloudformation', 'list_stacks', 'StackSummaries', 'NextToken'):
+            yield batch
+
+
+async def resync_cloudcontrol(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    is_global = is_global_resource(kind)
+    async for session in _get_sessions(None, DEFAULT_REGION if is_global else None):
+        region = session.region_name
+        logger.info(f"Resyncing {kind} in region {region}")
+        account_id = await find_account_id_by_session(session)
+        next_token = None
+        while True:
+            all_instances = []
+            try:
+                async with session.client("cloudcontrol") as cloudcontrol:
+                    params = {
+                        'TypeName': kind,
+                    }
+                    if next_token:
+                        params['NextToken'] = next_token
+                    
+                    response = await cloudcontrol.list_resources(**params)
+                    next_token = response.get('NextToken')
+                    resources = response.get('ResourceDescriptions', [])
+                    if not resources:
+                        break
+                    for instance in response.get('ResourceDescriptions', []):
+                        described = await describe_single_resource(kind, instance.get('Identifier'), account_id, region)
+                        described.update({KIND_PROPERTY: kind, ACCOUNT_ID_PROPERTY: account_id, REGION_PROPERTY: region})
+                        all_instances.append(_fix_unserializable_date_properties(described))
+                    yield all_instances
+            except Exception as e:
+                logger.error(f"Failed to list CloudControl Instance in account {account_id} kind {kind} region: {region}; error {e}")
+                break
+            if not next_token:
+                break
+                
+
+@ocean.on_resync(kind=ResourceKindsWithSpecialHandling.EC2)
+async def resync_ec2(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    async for session in _get_sessions():
+        region = session.region_name
+        account_id = await find_account_id_by_session(session)
+        try:
+            async with session.resource('ec2') as ec2:
+                async with session.client('ec2') as ec2_client:
+                    async for page in ec2.instances.pages():
+                        page_instances = []
+                        for instance in page:
+                            described_instance = await ec2_client.describe_instances(InstanceIds=[instance.id])
```

Comment:
its going to very very slow to describe each of them

Reply:
Yes but it's either that or losing properties